src/core/core.scale.js

```diff
@@ -659,7 +661,7 @@ module.exports = function(Chart) {
 			helpers.each(ticks, function(tick, index) {
 				var label = tick.label;
 				var lineWidth, lineColor, borderDash, borderDashOffset;
-				if (index === (typeof me.zeroLineIndex !== 'undefined' ? me.zeroLineIndex : 0)) {
+				if (index === (typeof me.zeroLineIndex !== 'undefined' ? me.zeroLineIndex : 0) && (options.offset === gridLines.offsetGridLines)) {
```

Comment:
This prevents the zero line from being drawn in case of 2. and 3. below.
<img width="416" alt="screen shot 2017-08-10 at 11 37 10 am" src="https://user-images.githubusercontent.com/723188/29152114-873ae356-7dc0-11e7-9b49-f926b5662433.png">